backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/creditLimit.feature

```diff
@@ -70,6 +70,9 @@ Feature: credit limit
       | C_BPartner_CreditLimit_ID.Identifier | Processed |
       | cl_1                                 | false     |
 
+    # we need to wait 1 second to make sure the first "create-shipment" work package - which failed - will not share the same second with the next "create-shipment" work package that we are about to enqueue
+    And wait 1s
```

Comment:
why can't it share the same second?

Reply:
I guess in reality we do have more than one sec between a failed processing, the release of the credit-stop and the next attempt..
Still..there might be other cases we we don't have htat much time between two retries

Reply:
it's related to `de.metas.async.service.AsyncBatchService#getWorkPackagesFromCurrentRun`, turns out we don't store milliseconds in `C_Queue_Workpackage` `Created` & `Updated` timestamp columns, so if the 1st attempt (which failed) and the 2nd one, run within the same second -> they will have the same created/update timestamp so the `getWorkPackagesFromCurrentRun` will consider both of them in the second run so the async batch fails... honestly, in a prod environment and real scenarios I don't think it will cause much trouble... after all, it's only a problem if that work package that's being wrongly considered it's a failed one...